client/trino-cli/src/main/java/io/trino/cli/MarkdownTablePrinter.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cli;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.trino.client.Column;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.repeat;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.client.ClientStandardTypes.BIGINT;
+import static io.trino.client.ClientStandardTypes.DECIMAL;
+import static io.trino.client.ClientStandardTypes.DOUBLE;
+import static io.trino.client.ClientStandardTypes.INTEGER;
+import static io.trino.client.ClientStandardTypes.REAL;
+import static io.trino.client.ClientStandardTypes.SMALLINT;
+import static io.trino.client.ClientStandardTypes.TINYINT;
+import static java.lang.Math.max;
+import static java.util.Objects.requireNonNull;
+import static org.jline.utils.AttributedString.stripAnsi;
+import static org.jline.utils.WCWidth.wcwidth;
+
+public class MarkdownTablePrinter
+        implements OutputPrinter
+{
+    private static final Set<String> NUMERIC_TYPES = ImmutableSet.of(TINYINT, SMALLINT, INTEGER, BIGINT, REAL, DOUBLE, DECIMAL);
+    private final List<String> fieldNames;
+    private final List<Align> alignments;
+    private final Writer writer;
+
+    private boolean headerOutput;
+
+    public MarkdownTablePrinter(List<Column> columns, Writer writer)
+    {
+        requireNonNull(columns, "columns is null");
+        this.fieldNames = columns.stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+        this.alignments = columns.stream()
+                .map(Column::getTypeSignature)
+                .map(signature -> NUMERIC_TYPES.contains(signature.getRawType()) ? Align.RIGHT : Align.LEFT)
+                .collect(toImmutableList());
+        this.writer = requireNonNull(writer, "writer is null");
+    }
+
+    private enum Align
+    {
+        LEFT,
+        RIGHT;
+    }
+
+    @Override
+    public void printRows(List<List<?>> rows, boolean complete)
+            throws IOException
+    {
+        int columns = fieldNames.size();
+
+        int[] maxWidth = new int[columns];
```

Comment:
nit: `columnWidth`

Reply:
This is copied from the aligned table printer. I wanted to keep them consistent, so it's easier to compare them. Should I apply such changes to the aligned table printer too?

Reply:
Leave it as it is. Thanks

Reply:
Note that this applies to the whole file